packages/block-editor/src/components/block-navigation/list.js

```diff
@@ -101,6 +75,88 @@ export default function BlockNavigationList( {
 				</li>
 			) }
 		</ul>
-		/* eslint-enable jsx-a11y/no-redundant-roles */
+	);
+}
+
+function NavigationBlock( { block, selectBlock, selectedBlockClientId, position, hasSiblings, showAppender, showBlockMovers, showNestedBlocks } ) {
+	const [ isHovered, setIsHovered ] = useState( false );
+	const [ isSelectionButtonFocused, setIsSelectionButtonFocused ] = useState( false );
+	const {
+		name,
+		clientId,
+		attributes,
+		innerBlocks,
+	} = block;
+	const blockType = getBlockType( name );
+	const blockDisplayName = getBlockDisplayName( blockType, attributes );
+
+	const wrapper = useRef( null );
+	const isSelected = clientId === selectedBlockClientId;
+	const adjustScrolling = false;
+	const enableAnimation = true;
+	const animateOnChange = position;
+
+	const style = useMovingAnimation( wrapper, isSelected, adjustScrolling, enableAnimation, animateOnChange );
+
+	return (
+		<animated.li ref={ wrapper } style={ style } role="treeitem">
+			<div
+				className={ classnames( 'editor-block-navigation__item block-editor-block-navigation__item', {
+					'is-selected': isSelected,
+				} ) }
+				onMouseEnter={ () => setIsHovered( true ) }
+				onMouseLeave={ () => setIsHovered( false ) }
+			>
+				<Button
+					className="editor-block-navigation__item-button block-editor-block-navigation__item-button"
+					onClick={ () => selectBlock( clientId ) }
+					onFocus={ () => setIsSelectionButtonFocused( true ) }
+					onBlur={ () => setIsSelectionButtonFocused( false ) }
+				>
+					<BlockIcon icon={ blockType.icon } showColors />
+					{ blockDisplayName }
+					{ isSelected && <span className="screen-reader-text">{ __( '(selected block)' ) }</span> }
+				</Button>
+				{ showBlockMovers && hasSiblings && (
+					<BlockMover
+						isHidden={ ! isHovered && ! isSelected && ! isSelectionButtonFocused }
+						clientIds={ [ clientId ] }
+					/>
+				) }
+			</div>
+			{ showNestedBlocks && !! innerBlocks && !! innerBlocks.length && (
+				<NavigationList
+					blocks={ innerBlocks }
+					selectedBlockClientId={ selectedBlockClientId }
+					selectBlock={ selectBlock }
+					showAppender={ showAppender }
+					showBlockMovers={ showBlockMovers }
+					showNestedBlocks={ showNestedBlocks }
+					parentBlockClientId={ clientId }
+				/>
+			) }
+		</animated.li>
+	);
+}
+
+export default function BlockNavigationList( props ) {
```

Comment:
Why not simply `export default NavigationList;`?

Reply:
I've recombined NavigationList and BlockNavigationList, so this should be resolved now.